kasa/iot/modules/emeter.py

```diff
@@ -4,130 +4,64 @@
     """Emeter module."""
 
-    def __init__(self, device: Device, module: str):
-        super().__init__(device, module)
-        self._add_feature(
-            Feature(
-                device,
-                name="Current consumption",
-                attribute_getter="current_consumption",
-                container=self,
-                unit="W",
-                id="current_power_w",  # for homeassistant backwards compat
-                precision_hint=1,
-                category=Feature.Category.Primary,
-            )
-        )
-        self._add_feature(
-            Feature(
-                device,
-                name="Today's consumption",
-                attribute_getter="emeter_today",
-                container=self,
-                unit="kWh",
-                id="today_energy_kwh",  # for homeassistant backwards compat
-                precision_hint=3,
-                category=Feature.Category.Info,
-            )
-        )
-        self._add_feature(
-            Feature(
-                device,
-                id="consumption_this_month",
-                name="This month's consumption",
-                attribute_getter="emeter_this_month",
-                container=self,
-                unit="kWh",
-                precision_hint=3,
-                category=Feature.Category.Info,
-            )
-        )
-        self._add_feature(
-            Feature(
-                device,
-                name="Total consumption since reboot",
-                attribute_getter="emeter_total",
-                container=self,
-                unit="kWh",
-                id="total_energy_kwh",  # for homeassistant backwards compat
-                precision_hint=3,
-                category=Feature.Category.Info,
-            )
-        )
-        self._add_feature(
-            Feature(
-                device,
-                name="Voltage",
-                attribute_getter="voltage",
-                container=self,
-                unit="V",
-                id="voltage",  # for homeassistant backwards compat
-                precision_hint=1,
-                category=Feature.Category.Primary,
-            )
-        )
-        self._add_feature(
-            Feature(
-                device,
-                name="Current",
-                attribute_getter="current",
-                container=self,
-                unit="A",
-                id="current_a",  # for homeassistant backwards compat
-                precision_hint=2,
-                category=Feature.Category.Primary,
-            )
-        )
+    _supported = (
+        EnergyInterface.ModuleFeature.CONSUMPTION_TOTAL
+        | EnergyInterface.ModuleFeature.PERIODIC_STATS
+        | EnergyInterface.ModuleFeature.VOLTAGE_CURRENT
+    )
+
+    def supports(self, module_feature: EnergyInterface.ModuleFeature) -> bool:
+        """Return True if module supports the feature."""
+        return self._supported & module_feature == module_feature
```

Comment:
The support checks should be done in the (base) interface.

Reply:
Done